MalApi/MyAnimeListApi.cs

```diff
@@ -72,6 +78,40 @@ private HttpRequestMessage InitNewRequest(string uri, HttpMethod method)
             return request;
         }
 
+        private HttpRequestMessage InitNewRequestWithCredentials(string user, string password, string uri, HttpMethod method, UpdateObjectBase updateInfo = null)
+        {
+            HttpRequestMessage request = new HttpRequestMessage(method, uri);
+
+            // Requests with credentials require them to be encoded in base64
+            string credentials;
+            if (user != null && password != null)
+            {
+                var plainTextBytes = Encoding.UTF8.GetBytes(user + ":" + password);
+                credentials =  Convert.ToBase64String(plainTextBytes);
+            }
+            else
+            {
+                throw new Exception("Invalid input credentials format (both username and password have to be input OR just the base64 representation");
+            }
+
+            if (UserAgent != null)
+            {
+                request.Headers.TryAddWithoutValidation("User-Agent", UserAgent);
```

Comment:
You can avoid repeating the logic that is in InitNewRequest by calling InitNewRequest and then adding onto the request it returns.